hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/TargetTypeManagement.java

```diff
@@ -0,0 +1,136 @@
+    Optional<TargetType> getByName(@NotEmpty String name);
+
+    /**
+     * @return total count
+     */
+    @PreAuthorize(SpPermission.SpringEvalExpressions.HAS_AUTH_READ_TARGET)
+    long count();
+
+    /**
+     * @param create
+     *          TargetTypeCreate
+     * @return targetType
+     */
+    @PreAuthorize(SpPermission.SpringEvalExpressions.HAS_AUTH_CREATE_TARGET)
+    TargetType create(@NotNull @Valid TargetTypeCreate create);
+
+    /**
+     * @param creates
+     *          List of TargetTypeCreate
+     * @return List of targetType
+     */
+    @PreAuthorize(SpPermission.SpringEvalExpressions.HAS_AUTH_CREATE_TARGET)
+    List<TargetType> create(@NotNull @Valid Collection<TargetTypeCreate> creates);
+
+    /**
+     * @param id targetTypeId
+     */
+    @PreAuthorize(SpPermission.SpringEvalExpressions.HAS_AUTH_DELETE_TARGET)
+    void delete(@NotNull Long id);
+
+    /**
+     * @param pageable
+     *          Page
+     * @return TargetType page
+     */
+    @PreAuthorize(SpPermission.SpringEvalExpressions.HAS_AUTH_READ_TARGET)
+    Slice<TargetType> findAll(@NotNull Pageable pageable);
+
+    /**
+     * @param pageable
+     *          Page
+     * @param rsqlParam
+     *          query param
+     * @return Target type
+     */
+    @PreAuthorize(SpPermission.SpringEvalExpressions.HAS_AUTH_READ_TARGET)
+    Page<TargetType> findByRsql(@NotNull Pageable pageable, @NotNull String rsqlParam);
+
+
+    /**
+     * @param id
+     *          Target type ID
+     * @return Target Type
+     */
+    @PreAuthorize(SpPermission.SpringEvalExpressions.HAS_AUTH_READ_TARGET)
+    Optional<TargetType> get(long id);
+
+
+    /**
+     * @param ids
+     *          List of Target type ID
+     * @return Target type list
+     */
+    @PreAuthorize(SpPermission.SpringEvalExpressions.HAS_AUTH_READ_TARGET)
+    List<TargetType> get(@NotEmpty Collection<Long> ids);
+
+
+    /**
+     * @param update
+     *          TargetTypeUpdate
+     * @return Target Type
+     */
+    @PreAuthorize(SpPermission.SpringEvalExpressions.HAS_AUTH_UPDATE_TARGET)
+    TargetType update(@NotNull @Valid TargetTypeUpdate update);
+
+    /**
+     * @param targetTypeId
+     *          Target type ID
+     * @param distributionSetTypeIds
+     *          Distribution set ID
+     * @return Target type
+     */
+    @PreAuthorize(SpPermission.SpringEvalExpressions.HAS_AUTH_READ_REPOSITORY_AND_UPDATE_TARGET)
+    TargetType assignCompatibleDistributionSetTypes(long targetTypeId,
+                                                    @NotEmpty Collection<Long> distributionSetTypeIds);
+
+    /**
+     * @param targetTypeId
+     *          Target type ID
+     * @param distributionSetTypeIds
+     *          Distribution set ID
+     * @return Target type
+     */
+    @PreAuthorize(SpPermission.SpringEvalExpressions.HAS_AUTH_READ_REPOSITORY_AND_UPDATE_TARGET)
+    TargetType unassignDistributionSetType(long targetTypeId, long distributionSetTypeIds);
+
```

Comment:
Add methods declarations that can make use of the Specifications suggested in the comment at https://github.com/eclipse/hawkbit/pull/1162/files#r690372459

_Suggestions:_
- `Optional<JpaTargetType> findByTargetId(long targetId)`
- `List<JpaTargetType> findByTargetIds(Collection<long> targetIds)`
- `Optional<JpaTargetType> findByTargetControllerId(String controllerId)`
- `List<JpaTargetType> findByTargetControllerIds(Collection<String> controllerIds)`